src/expr/src/expr/expr_regexp_count.rs

```diff
@@ -0,0 +1,252 @@
+                    )
+                    .map_err(|e| ExprError::Internal(e.into()))?;
+
+                    match start_datum {
+                        Some(ScalarImpl::Int32(start)) => {
+                            if start <= 0 {
+                                bail!("start must greater than zero");
+                            }
+                            Some(start as u32)
+                        }
+                        _ => bail!("Expected start to be a Unsigned Int32"),
+                    }
+                }
+                _ => {
+                    return Err(ExprError::UnsupportedFunction(
+                        "non-constant start in `regexp_count`".to_string(),
+                    ))
+                }
+            };
+
+            // See if `flags` is specified
+            if let Some(flags_node) = children.next() {
+                flags = match &flags_node.get_rex_node()? {
+                    RexNode::Constant(flags_value) => {
+                        let flags_datum = deserialize_datum(
+                            flags_value.get_body().as_slice(),
+                            &DataType::from(flags_node.get_return_type().unwrap()),
+                        )
+                        .map_err(|e| ExprError::Internal(e.into()))?;
+
+                        match flags_datum {
+                            Some(ScalarImpl::Utf8(flags)) => Some(flags.to_string()),
+                            _ => bail!("Expected flags to be a String"),
+                        }
+                    }
+                    _ => {
+                        return Err(ExprError::UnsupportedFunction(
+                            "non-constant flags in `regexp_count`".to_string(),
+                        ))
+                    }
+                }
+            }
+        };
+
+        // Sanity check
+        if children.next().is_some() {
+            bail!("syntax error in `regexp_count`");
+        }
+
+        let flags = if let Some(f) = flags {
+            f
+        } else {
+            "".to_string()
+        };
+
+        if flags.contains('g') {
+            bail!("`regexp_count` does not support global flag option");
+        }
+
+        let ctx = RegexpContext::new(&pattern, &flags)?;
+
+        Ok(Self {
+            source,
+            ctx,
+            return_type,
+            start,
+        })
+    }
+}
+
+impl RegexpCountExpression {
+    fn match_row(&self, text: Option<&str>) -> Option<i32> {
+        if let Some(text) = text {
+            // First get the start position to count for
+            let start = if let Some(s) = self.start { s - 1 } else { 0 };
+
+            // For unicode purpose
+            let mut start = match text.char_indices().nth(start as usize) {
+                Some((idx, _)) => idx,
+                // The start is out of bound or just invalid
+                // FIXME: What to return?
+                None => return None,
```

Comment:
the start is greater than or equal the char length, just return 0 match.
```suggestion
                // The start is out of bound
                None => return Some(0),
```